Integrate chapter about systematic uncertainties into docs (#579)
columnflow/columnflow: docs/user_guide/shifts.md

As demonstrated in the cms analysis example, this can be done like so:

```{literalinclude} ../../analysis_templates/cms_minimal/__cf_module_name__/config/analysis___cf_short_name_lc__.py
:dedent:
:language: python
:start-at: '# event weights due to muon scale factors'
:end-before: '# add column aliases for shift mu'
```

Next, we define a column alias as auxiliary information for each of these shifts.
This can later be used to switch out the name of a given column (e.g. `muon_weight`) with a varied version of the quantity (e.g. `muon_weight_up`).
For convenience, we provide the {py:func}`~columnflow.config_util.add_shift_aliases` function to easily add the alias for both variations of the shift at once:

```{literalinclude} ../../analysis_templates/cms_minimal/__cf_module_name__/config/analysis___cf_short_name_lc__.py
:dedent:
:language: python
:start-at: '# add column aliases for shift mu'
:end-before: '# external files'
```

Note that this function makes some assumptions on the naming of the varied quantities and the shifts.

Finally, we can now configure the modules within the workflow to use this shift.

> As mentioned above, this class of systematic uncertainties does not change any selection efficiencies.

Reviewer: The selection efficiency can be modified via event weights, but we can still reuse SelectEvents because the acceptance of any individual event is not changed.

```suggestion
As mentioned above, this class of systematic uncertainties does not affect the selection acceptance.
```

Therefore, it is sufficient to consider the varied quantities in the {py:class}`~columnflow.tasks.histograms.CreateHistograms` task.

> The calculation of the final event weight is handles by the {py:class}`~columnflow.weight.WeightProducer` instance you specify at command-line level.

Reviewer 2: ```suggestion
The calculation of the final event weight is handled by the {py:class}`~columnflow.weight.WeightProducer` instance you specify at command-line level.
```

This way, the workflow is now fully aware of the shift.
You can test your implementation e.g. by running the {py:class}`~columnflow.tasks.plotting.PlotShiftedVariables1D` task.
For more information, see the [plotting overview](../task_overview/plotting_tasks.md).

To include this shift in a statistical inference model, we need to add it as a parameter (see CMS analysis example):

```{literalinclude} ../../analysis_templates/cms_minimal/__cf_module_name__/inference/example.py
:dedent:
:language: python
:start-at: '# muon weight uncertainty'
:end-before: '# jet energy correction uncertainty'
```

In particular, note that this type of uncertainty needs to define the keyword argument `config_shift_source` in the {py:func}`~columnflow.inference.InferenceModel.add_parameter` function to properly resolve the necessary shifts to run through the workflow.

## Uncertainties that modify the selection efficiency

A more complex type of shape uncertainty arises from effects that can modify vital quantities for analysis decisions.
One example for this class of uncertainties are calibrations of objects such as jets, which modify the four-momenta content of these objects.

> Since these calibrations are generally subject to uncertainties, the corresponding variations can propagate throughout the analysis workflow and modify for example selection efficiencies in the analysis phase space.

Reviewer: ```suggestion
Since these calibrations are generally subject to uncertainties, the corresponding variations can propagate throughout the analysis workflow and modify event and object selection acceptances.
```

As explained in the {ref}`previous section <weight_based_uncertainties>`, the shifts need to be registered in the config instance:

```{literalinclude} ../../analysis_templates/cms_minimal/__cf_module_name__/config/analysis___cf_short_name_lc__.py
:dedent:
:language: python
:start-at: '# fake jet energy correction shift, with aliases flaged as "selection_dependent", i.e. the aliases'
:end-before: "# add column aliases for shift jec"
```

Note that in this case, we have also applied a `tag` to the shifts, which we can also later use to extract information about (groups of) uncertainties.

Next, the aliases for the varied columns are added in accordance with the naming scheme that the Calibrator module uses:

```{literalinclude} ../../analysis_templates/cms_minimal/__cf_module_name__/config/analysis___cf_short_name_lc__.py
:dedent:
:language: python
:start-at: "# add column aliases for shift jec"
:end-before: '# event weights due to muon scale factors'
```

Note that as opposed to the previous example, we use the `name` of the shift to construct the final name of the column containing the varied values.

Finally, we need to make the relevant modules aware of the shift.

> As explained above, the jet energy calibration and its variations can have a non-trivial effect on the selection efficiency in the final analysis phase space.

Reviewer: ```suggestion
As explained above, the jet energy calibration and its variations can have a non-trivial effect on the selection acceptance in the final analysis phase space.
```
